integration-test-tool/src/integration-tests.ts

```diff
@@ -0,0 +1,437 @@
+      // expect that clicking people works
+      await expect(page).toClick('.nav-link', {text: 'People'})
+      await expect(page).toMatchElement('h1', {text: 'Search People'})
+      // expect that we are logged in (therefore the button says Log Out)
+      await expect(page).toMatch('Log Out')
+      // click search
+      await expect(page).toClick('input[type="submit"][value="Search"]')
+
+      // wait for the response
+      await page.waitForResponse((req) => {
+        return req.url().includes('/people')
+      })
+      await page.waitFor(500)
+
+      // get the results view
+      const resultsPre = await page.$('.results-view')
+      const resultsText = await page.evaluate((pre) => pre.textContent, resultsPre) as string
+      await resultsPre.dispose()
+
+      // parse it
+      const resultsObj = JSON.parse(resultsText)
+      // expect it to contain no errors
+      expect(resultsObj.error).toBeUndefined()
+      // and some data (check the primary id field length)
+      expect(resultsObj.nconst.length).toBeGreaterThan(0)
+    })
+
+    it('GET /titles', async () => {
+      const res = await fetch(`${config.backend_site}/titles`, {
+        headers: {
+          Authorization: `Bearer ${idToken}`,
+        },
+        method: 'GET',
+      })
+
+      const status = res.status
+      const data = await res.json()
+
+      expect(status).toBe(200)
+      expect(data._embedded.titles.length).toBeGreaterThan(0)
+    })
+
+    describe('Backend titles', () => {
+      /**
+       * The first title in the titles
+       * Note: We populate this in the first test below, and reference it for the others
+       */
+      let firstTitle: any
+
+      beforeAll(async () => {
+        const res = await fetch(`${config.backend_site}/titles`, {
+          headers: {
+            Authorization: `Bearer ${idToken}`,
+          },
+          method: 'GET',
+        })
+
+        const data = await res.json()
+
+        firstTitle = data._embedded.titles[0]
+      })
+
+      it('GET /title', async () => {
+        const res = await fetch(`${config.backend_site}/titles/${firstTitle.tconst}`, {
+          headers: {
+            Authorization: `Bearer ${idToken}`,
+          },
+          method: 'GET',
+        })
+
+        const status = res.status
+        const data = await res.json()
+
+        expect(status).toBe(200)
+        expect(data).toEqual(firstTitle)
+      })
+
+      describe('POST /title', () => {
+        // tslint:disable:object-literal-sort-keys
+        const obj = {
+          tconst: 'inttest01',
+          titleType: 'short',
+          primaryTitle: 'integration test 01',
+          originalTitle: 'integration test 01',
+          isAdult: false,
+          startYear: 1337,
+          endYear: null,
+          runtimeMinutes: 1337,
+          genres: [
+            'test',
+            'integration',
+          ],
+        }
+        // tslint:enable:object-literal-sort-keys
```

Comment:
is this a comment or a way to set a setting in typescript?

Reply:
it controls the linter